### PR TITLE
[libmariadb] add zstd to option list

### PR DESCRIPTION
**Pull request by kotori2**
**Describe the pull request**

- #### What does your PR fix?
  `zstd` is required to build libmariadb. As it is bundled into macOS, it will not showing any error if you build it natively. But it will fail if you are cross compiling. 
```
[70/91] : && /Applications/Xcode_13.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -fPIC -g -arch arm64 -isysroot /Applications/Xcode_13.4.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -bundle -Wl,-headerpad_max_install_names  -o zstd.so CMakeFiles/zstd.dir/plugins/compress/c_zstd.c.o  /usr/local/lib/libzstd.dylib && :
FAILED: zstd.so 
: && /Applications/Xcode_13.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -fPIC -g -arch arm64 -isysroot /Applications/Xcode_13.4.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -bundle -Wl,-headerpad_max_install_names  -o zstd.so CMakeFiles/zstd.dir/plugins/compress/c_zstd.c.o  /usr/local/lib/libzstd.dylib && :
ld: warning: ignoring file /usr/local/lib/libzstd.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
  "_ZSTD_compressCCtx", referenced from:
      _ma_zstd_compress in c_zstd.c.o
  "_ZSTD_createCCtx", referenced from:
      _ma_zstd_ctx_init in c_zstd.c.o
  "_ZSTD_createDCtx", referenced from:
      _ma_zstd_ctx_init in c_zstd.c.o
  "_ZSTD_decompressDCtx", referenced from:
      _ma_zstd_decompress in c_zstd.c.o
  "_ZSTD_freeCCtx", referenced from:
      _ma_free_ctx in c_zstd.c.o
  "_ZSTD_freeDCtx", referenced from:
      _ma_free_ctx in c_zstd.c.o
  "_ZSTD_isError", referenced from:
      _ma_zstd_compress in c_zstd.c.o
      _ma_zstd_decompress in c_zstd.c.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Tested on `x64-osx` and cross compiling `arm64-osx`, other triplets should be not effected. 
  No because they are all skipped. 

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by kotori2** [comment]
> _This is a new experimental fast check for PR issues. Please let us know if this bot is helpful!_
> 
> PRs must add only one version and must not modify any published versions

I'm actually confused about this...

**Comment by dg0yt** [comment]
> I'm actually confused about this...

Maybe this was just fixed with #26930.

**Comment by dg0yt** [comment]
I think we must either generally depend on zstd or actively control zstd detection in the CMake project. 
Otherwise the actual dependencies / binary artifacts depend on installation order.

**Comment by kotori2** [comment]
> I think we must either generally depend on zstd or actively control zstd detection in the CMake project.
> Otherwise the actual dependencies / binary artifacts depend on installation order.

Sorry that I missed you reply. Do you mean we should add zstd to all platforms?

**Comment by vicroms** [comment]
[These lines are the problem](https://github.com/mariadb-corporation/mariadb-connector-c/blob/72b40bfaa869f3fe84242471dda989d13983d84c/plugins/compress/CMakeLists.txt#L15-L23):

```
IF(${ZSTD_FOUND})
  INCLUDE_DIRECTORIES(${ZSTD_INCLUDE_DIRS})
  REGISTER_PLUGIN(TARGET zstd
                  TYPE MARIADB_CLIENT_COMPRESSION_PLUGIN
                  CONFIGURATIONS DYNAMIC STATIC OFF
                  DEFAULT DYNAMIC
                  SOURCES ${COMPRESS_PLUGIN_DIR}/c_zstd.c
                  LIBRARIES ${ZSTD_LIBRARIES})
ENDIF()
```

We should patch them to 1) make `zstd` required and always build the compress plug-in or 2) introduce a port feature that explicitly controls the plugin and the `zstd` dependency.

**Comment by kotori2** [comment]
@vicroms Thanks for pointing out the issue. This PR is ready for review.

File: ports/libmariadb/portfile.cmake [hunk 1]
```diff
@@ -49,7 +51,8 @@ vcpkg_cmake_configure(
         -DWITH_SSL=${WITH_SSL}
         -DREMOTEIO_PLUGIN_TYPE=OFF
         -DAUTH_GSSAPI_PLUGIN_TYPE=OFF
-    MAYBE_UNUSED_VARIABLES
+        -DWITH_ZSTD=${WITH_ZSTD}
+        MAYBE_UNUSED_VARIABLES
```
Comment (dg0yt) [inline ports/libmariadb/portfile.cmake:55]: Please restore the indentation of the `MAYBE_UNUSED_VARIABLES` keyword.

**Review by dg0yt — CHANGES_REQUESTED** [review]
Looks good to me, but there is a small issue. Thanks.

File: ports/libmariadb/portfile.cmake (hunk 1 above)
Comment (kotori2) [reply ports/libmariadb/portfile.cmake]: Done. I didn't noticed that it was a keyword.

**Review by dg0yt — APPROVED** [review]

**Pull request merged**